content/docs/userspace/full-stack/6-eyre.md

```diff
@@ -0,0 +1,79 @@
++++
+title = "6. Eyre"
+weight = 6
+template = "doc.html"
++++
+
+Now that we have our structure file, agent, `$json` conversion library and mark
+file, our back-end is complete. Before we start writing our front-end, though,
+we should give a brief overview of how Eyre works.
+
+[Eyre](/docs/arvo/eyre/eyre) is the HTTP server [vane](/docs/glossary/vane) of
+Arvo. Eyre has a handful of different subsystems, but the main two are the
+channel system and the scry interface. These two are what we'll focus on here.
+
+In order to use the channel system or perform scries, a web client must have
+authenticated with the ship's web login code (e.g.
+`lidlut-tabwed-pillex-ridrup`) and obtained a session cookie. Our front-end will
+be served directly from the ship by the `%docket` agent, so we can assume a
+session cookie was already obtained when the user logged into landscape, and
+skip over authentication.
+
+## Channels
+
+Eyre's channel system is the main way to interact with agents from a web client.
+It provides a JSON interface to the ordinary poke and subscription system for
+Gall agents.
+
+First, a unique channel ID is generated by the web client (`@urbit/http-api`
+uses the current Unix time concatenated with a random hex string). The client
+then sends a poke or subscription request for the channel, and Eyre
+automatically opens a new channel with that ID. Once open, the client can then
+connect to the channel and receive any events such as poke acks, watch acks,
+facts from subscriptions, etc.
+
+The new channel is an SSE ([Server Sent
+Event](https://html.spec.whatwg.org/#server-sent-events)) stream, and can be
+handled by an `EventSource` object in Javascript. The `@urbit/http-api` library
+we'll use abstracts this for us, so we won't need to deal with an `EventSource`
+object directly. The channel can handle multiple concurrent subscriptions to
+different agent subscription paths, and different agents can be poked through
+the one channel. This means a client only needs to open a single channel for all
+of its interactions with the ship. Each subscription is given a different ID, so
+they can be individually unsubscribed later.
+
+A channel will timeout after 12 hours of inactivity, and the timeout is reset
+every time Eyre receives a message of any kind from the client. Additionally,
+each subscription on the channel may only accumulate 50 unacknowledged facts
+before it's considered "clogged", in which case the individual clogged
+subscription will be closed by Eyre after a short delay. All events of any kind
```

Comment:
is there any warning about data handling that needs to be parenthetically included here?  (because of clogged drops)

Reply:
i.e. client-side data handling

Reply:
no `http-api` automatically acks all events, so it's not really something to worry about when using the npm module except when thinking about disconnects/reconnects, and I've given some extra commentary about that later